trace/trace.go

```diff
@@ -0,0 +1,125 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/sunfmin/reflectutils"
+	kitlog "github.com/theplant/appkit/log"
+	"github.com/theplant/prottp"
+	"google.golang.org/grpc"
+)
+
+// UnaryServerInterceptor new a server interceptor
+func UnaryServerInterceptor(log kitlog.Logger, traces []MethodTrace) grpc.UnaryServerInterceptor {
+	for i := range traces {
+		// MethodName: GetArticleById
+		// FullMethod: /theplant.ec.api.blog.BlogService/GetArticleById
+		traces[i].MethodName = fmt.Sprintf("/%s", traces[i].MethodName)
+	}
+
+	var findTrace = func(name string) *MethodTrace {
```

Comment:
There's no strong need for `findTrace` to be defined per-interceptor. You could lift it out to the top level and pass in `traces` as a parameter.

Reply:
Changed.